apps/emqx_bridge/test/emqx_bridge_v2_api_SUITE.erl

```diff
@@ -1908,3 +1917,48 @@ t_kind_dependencies(Config) when is_list(Config) ->
         []
     ),
     ok.
+
+%% Verifies that we return thrown messages as is to the API.
+t_thrown_messages(matrix) ->
+    [
+        [single, actions]
+    ];
+t_thrown_messages(Config) when is_list(Config) ->
+    meck:expect(?CONNECTOR_IMPL, on_remove_channel, fun(_ConnResId, ConnState, _ActionResid) ->
+        timer:sleep(20_000),
+        {ok, ConnState}
+    end),
+    ?check_trace(
+        begin
+            ActionType = ?SOURCE_TYPE,
+            ConnectorType = ?SOURCE_CONNECTOR_TYPE,
+            ConnectorName = <<"c">>,
+            ActionName1 = <<"a1">>,
+            {ok, {{_, 201, _}, _, _}} =
+                emqx_bridge_v2_testlib:create_connector_api([
+                    {connector_config, source_connector_create_config(#{})},
+                    {connector_name, ConnectorName},
+                    {connector_type, ConnectorType}
+                ]),
+            CreateConfig = mqtt_action_create_config(#{
+                <<"connector">> => ConnectorName
+            }),
+            {201, _} = create_action_api(
+                ActionName1,
+                ActionType,
+                CreateConfig
+            ),
+            UpdateConfig = maps:remove(<<"type">>, CreateConfig),
+            ?assertMatch(
+                {400, #{<<"message">> := <<"Timed out trying to remove", _/binary>>}},
```

Comment:
If it is timeout, it should return 5xx since it is server side error, 4xx means client's fault.

Reply:
Makes sense.  Though, when handling thrown messages, it's tricky to distinguish between user and server errors.  I'll try to adjust that.  :thinking: